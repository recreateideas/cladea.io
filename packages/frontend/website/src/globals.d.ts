import { Theme } from "./App/ui-core/theme";
import { Store } from "redux";

export declare global {
  interface Window {
    __REDUX_DEVTOOLS_EXTENSION_COMPOSE__?: any;
    /** enzyme global methods for testing */
    shallow: any;
    render: any;
    mount: any;
    theme: Theme;
    ResizeObserver: any;
    store: Store;
  }
}

declare module "ui-core";
