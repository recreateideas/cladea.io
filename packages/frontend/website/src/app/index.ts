export * from "./App";
export * from "./features";
