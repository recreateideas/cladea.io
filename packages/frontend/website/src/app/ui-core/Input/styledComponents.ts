import styled from 'styled-components';
import { CustomCss } from '@types';

export const Container = styled.div<{ customCss?: CustomCss }>`
    width: 100%;
    display: flex;
    flex-direction: column;
    justify-content: center;
    ${(props) => props.customCss}
`;
