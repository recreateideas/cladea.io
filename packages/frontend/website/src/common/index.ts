export * from './utils';
export * from './hooks';
export * from './converters';
