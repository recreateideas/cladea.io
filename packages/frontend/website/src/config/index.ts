export const apiGatewaUrl = process.env.REACT_APP_API_GATEWAY_URL;

console.log(process.env);
