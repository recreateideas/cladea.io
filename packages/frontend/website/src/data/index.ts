import land110m from "./land-110m.json";

export { land110m };
