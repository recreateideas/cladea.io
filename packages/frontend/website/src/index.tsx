import "./app/ui-core/theme/GlobalStyle/roboto.css";
import ReactDOM from "react-dom";
import { Provider } from "react-redux";
import { store } from "./redux";
import { ServicedApp, ThemeProviderConnected } from "./app";

ReactDOM.render(
  <Provider store={store}>
    <ThemeProviderConnected>
      <ServicedApp />
    </ThemeProviderConnected>
  </Provider>,
  document.getElementById("root")
);
