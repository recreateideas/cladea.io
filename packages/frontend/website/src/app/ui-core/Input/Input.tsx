import { ReactElement, forwardRef, Ref } from 'react';
import { TextField, StandardTextFieldProps } from '@mui/material';
import { Container } from './styledComponents';
import { CustomCss } from '@types';

interface IProps extends StandardTextFieldProps {
    customCss?: CustomCss;
}

export const Input = forwardRef((props: IProps, ref: Ref<any>): ReactElement => {
    const { customCss, variant = 'standard', ...inputProps } = props;
    return (
        <Container className="input-container" {...{ customCss }}>
            <TextField ref={ref} {...inputProps} variant={variant} />
        </Container>
    );
});

Input.displayName = 'Input';
