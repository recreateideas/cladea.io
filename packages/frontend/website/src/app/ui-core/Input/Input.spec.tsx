import { Input } from "./Input";
import { TestProvider } from "src/tests";

const { mountWithProviderAndTheme } = TestProvider();
describe("<Input/>", () => {
  it("should render", () => {
    const state = {
      common: {},
    };
    const root = mountWithProviderAndTheme(<Input />, state);
    expect(root).toMatchSnapshot();
  });
});
