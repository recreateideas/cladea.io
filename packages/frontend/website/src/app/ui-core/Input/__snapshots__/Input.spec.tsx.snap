// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input/> should render 1`] = `
Array [
  .c1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  min-width: 0;
  padding: 0;
  margin: 0;
  border: 0;
  vertical-align: top;
}

.c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

@media (hover:none) {

}

<div
    className="c0 input-container"
  >
    <div
      className="c1 MuiFormControl-root MuiTextField-root-gGZDs MuiTextField-root"
    >
      Array [
        "",
        .c1 {
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.4375em;
  -webkit-letter-spacing: 0.00938em;
  -moz-letter-spacing: 0.00938em;
  -ms-letter-spacing: 0.00938em;
  letter-spacing: 0.00938em;
  color: #031A51;
  box-sizing: border-box;
  position: relative;
  cursor: text;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1.Mui-disabled {
  color: rgba(0,0,0,0.38);
  cursor: default;
}

.c4 {
  font: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  color: currentColor;
  padding: 4px 0 5px;
  border: 0;
  box-sizing: content-box;
  background: none;
  height: 1.4375em;
  margin: 0;
  -webkit-tap-highlight-color: transparent;
  display: block;
  min-width: 0;
  width: 100%;
  -webkit-animation-name: mui-auto-fill-cancel;
  animation-name: mui-auto-fill-cancel;
  -webkit-animation-duration: 10ms;
  animation-duration: 10ms;
}

.c4::-webkit-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
}

.c4::-moz-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
}

.c4:-ms-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
}

.c4::-ms-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
}

.c4:focus {
  outline: 0;
}

.c4:invalid {
  box-shadow: none;
}

.c4::-webkit-search-decoration {
  -webkit-appearance: none;
}

label[data-shrink=false] + .MuiInputBase-formControl .c3::-webkit-input-placeholder {
  opacity: 0 !important;
}

label[data-shrink=false] + .MuiInputBase-formControl .c3::-moz-placeholder {
  opacity: 0 !important;
}

label[data-shrink=false] + .MuiInputBase-formControl .c3:-ms-input-placeholder {
  opacity: 0 !important;
}

label[data-shrink=false] + .MuiInputBase-formControl .c3::-ms-input-placeholder {
  opacity: 0 !important;
}

label[data-shrink=false] + .MuiInputBase-formControl .c3:focus::-webkit-input-placeholder {
  opacity: 0.42;
}

label[data-shrink=false] + .MuiInputBase-formControl .c3:focus::-moz-placeholder {
  opacity: 0.42;
}

label[data-shrink=false] + .MuiInputBase-formControl .c3:focus:-ms-input-placeholder {
  opacity: 0.42;
}

label[data-shrink=false] + .MuiInputBase-formControl .c3:focus::-ms-input-placeholder {
  opacity: 0.42;
}

.c4.Mui-disabled {
  opacity: 1;
  -webkit-text-fill-color: rgba(0,0,0,0.38);
}

.c4:-webkit-autofill {
  -webkit-animation-duration: 5000s;
  animation-duration: 5000s;
  -webkit-animation-name: mui-auto-fill;
  animation-name: mui-auto-fill;
}

.c2 {
  position: relative;
}

label + .c0 {
  margin-top: 16px;
}

.c2:after {
  border-bottom: 2px solid #12D8F7;
  left: 0;
  bottom: 0;
  content: "";
  position: absolute;
  right: 0;
  -webkit-transform: scaleX(0);
  -ms-transform: scaleX(0);
  transform: scaleX(0);
  -webkit-transition: -webkit-transform 200ms cubic-bezier(0.0,0,0.2,1) 0ms;
  -webkit-transition: transform 200ms cubic-bezier(0.0,0,0.2,1) 0ms;
  transition: transform 200ms cubic-bezier(0.0,0,0.2,1) 0ms;
  pointer-events: none;
}

.c2.Mui-focused:after {
  -webkit-transform: scaleX(1);
  -ms-transform: scaleX(1);
  transform: scaleX(1);
}

.c2.Mui-error:after {
  border-bottom-color: #d32f2f;
  -webkit-transform: scaleX(1);
  -ms-transform: scaleX(1);
  transform: scaleX(1);
}

.c2:before {
  border-bottom: 1px solid rgba(0,0,0,0.42);
  left: 0;
  bottom: 0;
  content: "\\00a0";
  position: absolute;
  right: 0;
  -webkit-transition: border-bottom-color 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: border-bottom-color 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
  pointer-events: none;
}

.c2:hover:not(.Mui-disabled):before {
  border-bottom: 2px solid #031A51;
}

.c2.Mui-disabled:before {
  border-bottom-style: dotted;
}

@media (hover:none) {
  .c2:hover:not(.Mui-disabled):before {
    border-bottom: 1px solid rgba(0,0,0,0.42);
  }
}

<div
          className="c0 c1 c2 MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="c3 MuiInput-input-tFIHl c4 MuiInput-input MuiInputBase-input"
            disabled={false}
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="text"
          />
        </div>,
      ]
    </div>
  </div>,
  "",
]
`;
